bot/cogs/team.py

```diff
@@ -0,0 +1,346 @@
+                                member.display_name
+                                + " was successfully removed from team "
+                                + t.name
+                                + ".",
+                            )
+                        except:
+                            logger.error(
+                                "Could not remove "
+                                + member.display_name
+                                + " from "
+                                + t.name
+                            )
+                            await reply_and_delete(
+                                ctx, "Error in removing member from team."
+                            )
+            if not removed:
+                await reply_and_delete(
+                    ctx, member.display_name + " was not in any team."
+                )
+
+    @team.command(
+        help="Move all connected users to a temporary voice channel based on their team using: !team voice"
+    )
+    @commands.has_permissions(administrator=True)
+    async def voice(self, ctx):
+        original_vc = ctx.author.voice.channel
+        for team in self.teams:
+            if not team.vc:
+                try:
+                    team.vc = await ctx.guild.create_voice_channel(
+                        team.name, category=original_vc.category
+                    )
+                except errors.Forbidden:
+                    await reply_and_delete(
+                        ctx,
+                        "The bot needs permission to manage channels and move members",
+                    )
+            for member in team.members:
+                if member.voice.channel == original_vc:
+                    try:
+                        await member.move_to(team.vc)
+                    except errors.Forbidden:
+                        await reply_and_delete(
+                            ctx,
+                            "The bot needs permission to manage channels and move members",
+                        )
+
+    @team.command(
+        help="Move all members in a team channel to given voice channel using: !team gather [Voice channel]"
+    )
+    @commands.has_permissions(administrator=True)
+    async def gather(self, ctx, vc: VoiceChannel):
+        for team in self.teams:
+            for member in team.vc.members:
+                await member.move_to(vc)
+        await reply_and_delete(ctx, "Successfully moved members")
+
+    @team.group(
+        name="divide",
+        invoke_without_command=True,
+        help="Get help for team divide commands",
+    )
+    async def divide(self, ctx):
+        await reply_and_delete(ctx, "'team divide' needs a sub-command!")
+        await ctx.send_help(self.team.divide)
+
+    @divide.command(
+        help="Split current voice channel into number groups using: !team divide random [number of groups]"
+    )
+    @commands.has_permissions(administrator=True)
+    async def random(self, ctx, n: str):
```

Comment:
```suggestion
    async def random(self, ctx, n: int):
```

This should force discord.py automatically convert the value for you

Reply:
I chose to cast the int myself to handle errors myself.